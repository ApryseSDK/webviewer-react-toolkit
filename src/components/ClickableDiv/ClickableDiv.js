'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.ClickableDiv = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var hooks_1 = require('../../hooks');
exports.ClickableDiv = (0, react_1.forwardRef)(function(_a, ref) {
  var onClick = _a.onClick,
    onKeyPress = _a.onKeyPress,
    disabled = _a.disabled,
    noFocusStyle = _a.noFocusStyle,
    usePointer = _a.usePointer,
    className = _a.className,
    children = _a.children,
    tabIndex = _a.tabIndex,
    divProps = tslib_1.__rest(_a, [
      'onClick',
      'onKeyPress',
      'disabled',
      'noFocusStyle',
      'usePointer',
      'className',
      'children',
      'tabIndex',
    ]);
  var clickableDivRef = (0, react_1.useRef)(null);
  (0, react_1.useImperativeHandle)(ref, function() {
    return clickableDivRef.current;
  });
  var handleOnClick = (0, hooks_1.useOnClick)(onClick, {
    disabled: disabled,
    stopPropagation: true,
  });
  var handleKeyPress = (0, hooks_1.useKeyForClick)(onKeyPress, clickableDivRef);
  var isUserTabbing = (0, hooks_1.useAccessibleFocus)();
  var clickableDivClass = (0, classnames_1.default)(
    'ui__base ui__clickableDiv',
    {
      'ui__clickableDiv--disabled': disabled,
      'ui__clickableDiv--tabbing': isUserTabbing,
      'ui__clickableDiv--noFocusStyle': noFocusStyle,
      'ui__clickableDiv--usePointer': usePointer && !disabled,
    },
    className,
  );
  return react_1.default.createElement(
    'div',
    tslib_1.__assign({}, divProps, {
      role: 'button',
      tabIndex: disabled
        ? -1
        : tabIndex !== null && tabIndex !== void 0
        ? tabIndex
        : 0,
      className: clickableDivClass,
      onClick: handleOnClick,
      onKeyPress: handleKeyPress,
      ref: clickableDivRef,
    }),
    children,
  );
});
//# sourceMappingURL=ClickableDiv.js.map
