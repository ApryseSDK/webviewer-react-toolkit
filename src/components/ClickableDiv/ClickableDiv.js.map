{"version":3,"file":"ClickableDiv.js","sourceRoot":"","sources":["ClickableDiv.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAuF;AACvF,qCAA6E;AAmBhE,QAAA,YAAY,GAAG,IAAA,kBAAU,EACpC,UAAC,EAAuG,EAAE,GAAG;IAA1G,IAAA,OAAO,aAAA,EAAE,UAAU,gBAAA,EAAE,QAAQ,cAAA,EAAE,YAAY,kBAAA,EAAE,UAAU,gBAAA,EAAE,SAAS,eAAA,EAAE,QAAQ,cAAA,EAAE,QAAQ,cAAA,EAAK,QAAQ,sBAArG,wGAAuG,CAAF;IACpG,IAAM,eAAe,GAAG,IAAA,cAAM,EAAiB,IAAI,CAAC,CAAC;IACrD,IAAA,2BAAmB,EAAC,GAAG,EAAE,cAAM,OAAA,eAAe,CAAC,OAAyB,EAAzC,CAAyC,CAAC,CAAC;IAE1E,IAAM,aAAa,GAAG,IAAA,kBAAU,EAAC,OAAO,EAAE,EAAE,QAAQ,UAAA,EAAE,eAAe,EAAE,IAAI,EAAE,CAAC,CAAC;IAE/E,IAAM,cAAc,GAAG,IAAA,sBAAc,EAAC,UAAU,EAAE,eAAe,CAAC,CAAC;IAEnE,IAAM,aAAa,GAAG,IAAA,0BAAkB,GAAE,CAAC;IAE3C,IAAM,iBAAiB,GAAG,IAAA,oBAAU,EAClC,2BAA2B,EAC3B;QACE,4BAA4B,EAAE,QAAQ;QACtC,2BAA2B,EAAE,aAAa;QAC1C,gCAAgC,EAAE,YAAY;QAC9C,8BAA8B,EAAE,UAAU,IAAI,CAAC,QAAQ;KACxD,EACD,SAAS,CACV,CAAC;IAEF,OAAO,CACL,0DACM,QAAQ,IACZ,IAAI,EAAC,QAAQ,EACb,QAAQ,EAAE,QAAQ,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,QAAQ,aAAR,QAAQ,cAAR,QAAQ,GAAI,CAAC,EACvC,SAAS,EAAE,iBAAiB,EAC5B,OAAO,EAAE,aAAa,EACtB,UAAU,EAAE,cAAc,EAC1B,GAAG,EAAE,eAAe,KAEnB,QAAQ,CACL,CACP,CAAC;AACJ,CAAC,CACF,CAAC"}