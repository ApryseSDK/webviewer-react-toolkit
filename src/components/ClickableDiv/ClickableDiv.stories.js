'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var ClickableDiv_1 = require('../ClickableDiv');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/ClickableDiv',
  component: ClickableDiv_1.ClickableDiv,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    ClickableDiv_1.ClickableDiv,
    {
      disabled: (0, addon_knobs_1.boolean)('disabled', false),
      noFocusStyle: (0, addon_knobs_1.boolean)('noFocusStyle', false),
      usePointer: (0, addon_knobs_1.boolean)('usePointer', false),
      onClick: (0, action_1.action)('onClick'),
    },
    (0, addon_knobs_1.text)('children', 'This is a clickable div!'),
  );
};
exports.Basic = Basic;
//# sourceMappingURL=ClickableDiv.stories.js.map
