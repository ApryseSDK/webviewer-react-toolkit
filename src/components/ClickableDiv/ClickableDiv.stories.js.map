{"version":3,"file":"ClickableDiv.stories.js","sourceRoot":"","sources":["ClickableDiv.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAAuD;AACvD,wDAA0B;AAC1B,gEAA+D;AAC/D,gDAA+C;AAC/C,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,yBAAyB,EAAE,SAAS,EAAE,2BAAY,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE9F,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,2BAAY,IACX,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,EACpC,YAAY,EAAE,IAAA,qBAAO,EAAC,cAAc,EAAE,KAAK,CAAC,EAC5C,UAAU,EAAE,IAAA,qBAAO,EAAC,YAAY,EAAE,KAAK,CAAC,EACxC,OAAO,EAAE,IAAA,eAAM,EAAC,SAAS,CAAC,IAEzB,IAAA,kBAAI,EAAC,UAAU,EAAE,0BAA0B,CAAC,CAChC,CAChB,EAT0B,CAS1B,CAAC;AATW,QAAA,KAAK,SAShB"}