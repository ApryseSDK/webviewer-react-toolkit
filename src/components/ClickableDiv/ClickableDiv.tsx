import classnames from 'classnames';
import React, { forwardRef, HTMLAttributes, MouseEvent, KeyboardEvent, useRef, useImperativeHandle } from 'react';
import { Omit } from '../../utils/typeUtils';

export interface ClickableDivProps extends Omit<HTMLAttributes<HTMLDivElement>, 'role' | 'tabIndex'> {
  /**
   * Is the clickable div disabled.
   */
  disabled?: boolean;
}

export const ClickableDiv = forwardRef<HTMLDivElement, ClickableDivProps>(
  ({ onClick, onKeyUp, disabled, className, children, ...buttonProps }, ref) => {
    const divRef = useRef<HTMLDivElement>(null);
    useImperativeHandle(ref, () => divRef.current!);

    const clickableDivClass = classnames('ui__base ui__clickableDiv', className, {
      ['ui__clickableDiv--disabled']: disabled,
    });

    const handleOnClick = (event: MouseEvent<HTMLDivElement>) => {
      if (!disabled) onClick?.(event);
    };

    const handleKeyUp = (event: KeyboardEvent<HTMLDivElement>) => {
      // Fire click on space press.
      if (event.key === ' ') divRef.current?.click();
      onKeyUp?.(event);
    };

    return (
      <div
        {...buttonProps}
        role="button"
        tabIndex={0}
        className={clickableDivClass}
        onClick={handleOnClick}
        onKeyUp={handleKeyUp}
        ref={divRef}
      >
        {children}
      </div>
    );
  },
);
