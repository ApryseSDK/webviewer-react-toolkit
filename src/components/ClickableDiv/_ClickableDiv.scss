.ui__clickableDiv {
  display: inline-block;

  &--disabled {
    opacity: 0.8;
  }
}
