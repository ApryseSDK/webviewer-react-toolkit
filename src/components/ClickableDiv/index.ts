export { ClickableDiv as default, ClickableDivProps } from './ClickableDiv';
