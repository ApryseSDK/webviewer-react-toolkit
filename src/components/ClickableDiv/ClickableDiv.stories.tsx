import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import React from 'react';
import ClickableDiv from '../ClickableDiv';
import docs from './README.md';

export default { title: 'ClickableDiv', component: ClickableDiv, parameters: { info: docs } };

export const basic = () => (
  <ClickableDiv
    disabled={boolean('disabled', false)}
    noFocusStyle={boolean('noFocusStyle', false)}
    usePointer={boolean('usePointer', false)}
    onClick={action('onClick')}
  >
    {text('children', 'This is a clickable div!')}
  </ClickableDiv>
);
