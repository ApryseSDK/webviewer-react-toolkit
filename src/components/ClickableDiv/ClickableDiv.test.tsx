import { mount, shallow } from 'enzyme';
import React from 'react';
import { spy } from 'sinon';
import ClickableDiv from '../ClickableDiv';

describe('ClickableDiv component', () => {
  it('renders its contents', () => {
    const clickableDiv = shallow(<ClickableDiv />);
    expect(clickableDiv.find('.ui__clickableDiv').length).toEqual(1);
  });

  it('snapshot renders default clickableDiv', () => {
    const clickableDiv = shallow(<ClickableDiv />);
    expect(clickableDiv).toMatchSnapshot();
  });

  it('clicking clickableDiv triggers onClick prop', () => {
    const onClick = spy();
    shallow(<ClickableDiv onClick={onClick} />).simulate('click');
    expect(onClick.callCount).toBe(1);
  });

  it('clicking disabled clickableDiv does not trigger onClick prop', () => {
    const onClick = spy();
    // full DOM mount so `clickableDiv` element will use disabled prop
    mount(<ClickableDiv onClick={onClick} disabled />).simulate('click');
    expect(onClick.callCount).toBe(0);
  });
});
