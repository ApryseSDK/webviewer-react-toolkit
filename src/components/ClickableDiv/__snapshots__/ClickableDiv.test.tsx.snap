// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableDiv component snapshot renders default clickableDiv 1`] = `
<div
  className="ui__clickableDiv"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
/>
`;
